onnxruntime/core/providers/cpu/tensor/upsample.cc

```diff
@@ -471,8 +456,8 @@ static BilinearParams SetupUpsampleBilinear(int64_t input_height,
   p.dx2 = p.dx1 + output_width;
 
   // Start processing
-  auto roi_y_start = roi.size() / 2 - 2;
-  auto roi_y_end = roi.size() - 2;
+  auto roi_y_start = is_nchw ? roi.size() / 2 - 2 : roi.size() / 2 - 3;
```

Comment:
>3 [](http://example.com/codeflow?start=69&length=1)

could you please give a name to this magic number, like width_idx?

Reply:
It is not width_idx and hard to find a magic number name for it. Would add a comment to explain it.

Reply:
It is the index of W in [N,H,W,C] or [N,C,H,W], which is mapped to corresponding idx in roi[start1, ...startN, end1, .., endN]?

Reply:
For [N,C,H,W], width_idx is 3. But the corresponding is  roi_y_start is roi.size() / 2 - 2. For [N,H,W,C], width_idx is 2. But the corresponding is  roi.size() / 2 - 3.

Reply:
Use width_ridx instead